===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ portset_add:n$2 == { -1 } }) free(*(portset_add:p)) at 3 (line 232, column 5) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/sfutil/ipobj.c
+++ b/table1/benchmarks/snort-2.9.13/src/sfutil/ipobj.c
@@ -227,7 +227,9 @@ static int portset_add(PORTSET * portset, unsigned port_lo, unsigned port_hi)
     p->port_lo = port_lo;
     p->port_hi = port_hi;
 
-    sflist_add_tail(&portset->port_list, p );
+    if (sflist_add_tail(&portset->port_list, p ) == -1)
+        free(p);
+
 
     return 0;
 }
